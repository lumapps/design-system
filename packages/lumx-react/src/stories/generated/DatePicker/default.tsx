../../../../../site-demo/content/product/components/date-picker/react/default.tsx