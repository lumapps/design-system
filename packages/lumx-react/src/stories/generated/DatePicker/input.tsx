/home/gcornut/git/dup-design-system/packages/site-demo/content/product/components/date-picker/react/input.tsx