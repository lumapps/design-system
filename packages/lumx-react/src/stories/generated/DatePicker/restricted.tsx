/home/gcornut/git/dup-design-system/packages/site-demo/content/product/components/date-picker/react/restricted.tsx