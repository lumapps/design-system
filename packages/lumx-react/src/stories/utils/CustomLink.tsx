/* eslint-disable jsx-a11y/anchor-has-content,jsx-a11y/anchor-is-valid */
import React from 'react';

/**
 * Example custom link that can be used in linkAs props.
 */
export const CustomLink: any = ({ ...props }) => <a href="#" data-custom-link {...props} />;
