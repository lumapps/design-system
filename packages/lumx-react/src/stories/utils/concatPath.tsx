export type PathPart = string | number;
export type Path = PathPart | Array<PathPart>;

/**
 * Concat flatten object path
 *
 * @example concatPath('foo', 'bar') // => 'foo.bar'
 * @example concatPath(['foo', 0]) // => 'foo[0]'
 * @example concatPath('foo', 0, ['bar']) // => 'foo[0].bar'
 */
export const concatPath = (...prefix: Path[]) => {
    const [first, ...rest] = prefix.flat();
    return rest.reduce<string>((acc, part) => {
        if (typeof part === 'number') return `${acc}[${part}]`;
        return `${acc}.${part}`;
    }, String(first));
};
