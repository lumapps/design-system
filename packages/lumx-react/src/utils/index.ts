/**
 * WARNING: All modules exported here are exposed to NPM in '@lumx/react/utils'.
 */

export { ClickAwayProvider } from './ClickAwayProvider';
export { Portal, type PortalProps, type PortalInit, PortalProvider, type PortalProviderProps } from './Portal';
