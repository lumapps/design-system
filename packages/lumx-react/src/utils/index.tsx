export * from '@lumx/core/js/utils';

export * from './getRootClassName';

export * from './htmlDecode';

export * from './partitionMulti';

export * from './type';
