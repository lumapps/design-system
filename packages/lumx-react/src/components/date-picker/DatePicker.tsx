import moment from 'moment';
import React, { useState } from 'react';
import { Comp } from '@lumx/react/utils';
import { CLASSNAME, COMPONENT_NAME } from './constants';
import { DatePickerControlled } from './DatePickerControlled';
import { DatePickerProps } from './types';

export const DatePicker: Comp<DatePickerProps> = ({ defaultMonth, locale, value, onChange, ...forwardedProps }) => {
    let castedValue;
    if (value) {
        castedValue = moment(value);
    } else if (defaultMonth) {
        castedValue = moment(defaultMonth);
    }
    if (castedValue && !castedValue.isValid()) {
        // eslint-disable-next-line no-console
        console.warn(`[@lumx/react/DatePicker] Invalid date provided ${castedValue}`);
    }
    const selectedDay = castedValue && castedValue.isValid() ? castedValue : moment();

    const [monthOffset, setMonthOffset] = useState(0);

    const setPrevMonth = () => setMonthOffset(monthOffset - 1);
    const setNextMonth = () => setMonthOffset(monthOffset + 1);

    const onDatePickerChange = (newDate?: Date) => {
        onChange(newDate);
        setMonthOffset(0);
    };

    const selectedMonth = moment(selectedDay).locale(locale).add(monthOffset, 'months').toDate();

    return (
        <DatePickerControlled
            {...forwardedProps}
            defaultMonth={defaultMonth}
            locale={locale}
            value={value}
            onPrevMonthChange={setPrevMonth}
            onNextMonthChange={setNextMonth}
            selectedMonth={selectedMonth}
            onChange={onDatePickerChange}
        />
    );
};
DatePicker.displayName = COMPONENT_NAME;
DatePicker.className = CLASSNAME;
