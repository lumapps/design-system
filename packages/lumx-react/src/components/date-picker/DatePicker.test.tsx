import React, { ReactElement } from 'react';

import { mount, shallow } from 'enzyme';
import 'jest-enzyme';

import { Wrapper } from '@lumx/react/testing/utils';
import { DatePickerProps } from '@lumx/react';
import { DatePicker } from './DatePicker';

const mockedDate = new Date(
    new Date(1487721600).toLocaleString('en-US', {
        timeZone: 'America/Toronto',
    }),
);
Date.now = jest.fn(() => mockedDate.valueOf());

type SetupProps = Partial<DatePickerProps>;

const setup = ({ ...propsOverride }: SetupProps = {}, shallowRendering = true) => {
    const props: DatePickerProps = {
        locale: 'fr',
        onChange: jest.fn(),
        value: mockedDate,
        nextButtonProps: { label: 'Next month' },
        previousButtonProps: { label: 'Previous month' },
        ...propsOverride,
    };
    const renderer: (el: ReactElement) => Wrapper = shallowRendering ? shallow : mount;
    const wrapper: Wrapper = renderer(<DatePicker {...props} />);

    return { props, wrapper };
};

describe(`<${DatePicker.displayName}>`, () => {
    describe('Snapshots and structure', () => {
        it('should render correctly', () => {
            const { wrapper } = setup();
            expect(wrapper).toMatchSnapshot();
        });
    });
});
