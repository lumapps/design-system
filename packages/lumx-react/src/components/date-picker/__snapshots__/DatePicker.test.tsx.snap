// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DatePicker> Snapshots and structure should render correctly 1`] = `
<DatePickerControlled
  locale="fr"
  nextButtonProps={
    Object {
      "label": "Next month",
    }
  }
  onChange={[Function]}
  onNextMonthChange={[Function]}
  onPrevMonthChange={[Function]}
  previousButtonProps={
    Object {
      "label": "Previous month",
    }
  }
  selectedMonth={1970-01-17T23:15:21.000Z}
  value={1970-01-17T23:15:21.000Z}
/>
`;
